### PR TITLE
Trip index

### PR DESCRIPTION
**Pull request by jamesrnelson**
###### Changes Made:
* Also added factory bot to test suite
* added index and show routes for trips
* added trips controller--index method contains pagination logic
* added trips index view that displays all attributes of each trip and only shows 30 trips per page.

### CODE REVIEW

File: app/views/trips/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    Start Station: <%= trip.start_station %> -
+    End Date: <%= trip.end_date %> -
+    End Station: <%= trip.end_station %> -
+    Bike ID: <%= trip.bike_id %> -
+    Subscription Type: <%= trip.subscription_type %> -
+    Zip Code: <%= trip.zip_code %>
+  </p>
+<% end %>
+
+<% @pages.times do |n| %>
```
[the remark this answers is not in the document]
Comment (anon0mys) [reply app/views/trips/index.html.erb]: I like page numbers personally.

File: spec/features/user_visits_trips_index_sees_first_30_trips_spec.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    expect(page).to have_content(trips.first.start_date)
+    expect(page).to have_content(trips.first.start_station)
+    expect(page).to have_content(trips.first.end_date)
+    expect(page).to have_content(trips.first.end_station)
+    expect(page).to have_content(trips.first.bike_id)
+    expect(page).to have_content(trips.first.subscription_type)
+    expect(page).to have_content(trips.first.zip_code)
+
+    expect(page).to have_content(trips[29].bike_id)
+    expect(page).to_not have_content(trips[30].bike_id)
```
Comment (anon0mys) [inline spec/features/user_visits_trips_index_sees_first_30_trips_spec.rb]: I really like this test structure. Test that one has all the attributes, then test that a couple random bikes are also there.

File: spec/models/trip_spec.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+require 'rails_helper'
+
+describe Trip, type: :model do
+  describe 'Validations' do
```
Comment (anon0mys) [inline spec/models/trip_spec.rb]: @jrambold do know if the trip data is clean? Should we validate all the attributes or do we want to allow for incomplete records?

File: app/views/trips/index.html.erb (hunk 1 above)
[the remark this answers is not in the document]
Comment (AliSchlereth) [reply app/views/trips/index.html.erb]: Agree about using numbers. Generally on the spec if something is in quotes, we are looking for that exact wording, if not in quotes you have more wiggle room.

**Pull request merged**